File: c++/11055.cpp
#include <algorithm>
#include <iostream>

using namespace std;

int arr[1000];
int dp[1000];

int main() {
  freopen("example.txt", "r", stdin);

  int N;
  cin >> N;

  for (int i = 0; i < N; i++) {
    cin >> arr[i];
  }

  for (int i = 0; i < N; i++) {
    int k = 0;
    for (int j = 0; j < i; j++) {
      if (arr[j] < arr[i]) k = max(k, dp[j]);
    }
    dp[i] = arr[i] + k;
  }

  int answer = 0;
  for (int i = 0; i < N; i++) {
    answer = max(answer, dp[i]);
  }
  cout << answer;

  return 0;
}

File: js/11055.js
const fs = require('fs');

const inputs = fs.readFileSync('example.txt').toString().trim().split('\n');

const N = +inputs[0];
const arr = inputs[1].split(' ').map(Number);

// dp[i] -> arr[i]를 포함하는 증가하는 부분 수열의 합
const dp = Array(N).fill(0);

for (let i = 0; i < N; i++) {
  max = 0;
  for (let j = 0; j < i; j++) {
    if (arr[j] < arr[i]) max = Math.max(max, dp[j]);
  }
  dp[i] = arr[i] + max;
}

console.log(Math.max(...dp));
